examples/twittermap/web/public/javascripts/map/controllers.js

```diff
@@ -691,30 +717,44 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common','clou
       },
 
       function(newResult, oldValue) {
-        if (newResult['mapResult'] !== oldValue['mapResult']) {
-          $scope.result = newResult['mapResult'];
-          if (Object.keys($scope.result).length !== 0) {
-            $scope.status.init = false;
-            drawMap($scope.result);
-          } else {
-            drawMap($scope.result);
-          }
-        }
-        if (newResult['totalCount'] !== oldValue['totalCount']) {
-          $scope.totalCount = newResult['totalCount'];
-        }
-        if(newResult['doNormalization'] !== oldValue['doNormalization']) {
-          $scope.doNormalization = newResult['doNormalization'];
-          drawMap($scope.result);
-        }
-        if(newResult['doSentiment'] !== oldValue['doSentiment']) {
-          $scope.doSentiment = newResult['doSentiment'];
-          if($scope.doSentiment) {
-            $scope.infoPromp = "Score";  // change the info promp
-          } else {
-            $scope.infoPromp = config.mapLegend;
-          }
-          drawMap($scope.result);
+        switch ($rootScope.maptype) {
```

Comment:
For this part, If the "mapResult" name will be used for all three types of map, I think "if (newResult['mapResult'] !== oldValue['mapResult']) {" should be placed first before switch. Since this watchCollection checks whether the mapResult is changed, then conducts some necessary operations. 

if (new !== old) {

switch ($rootScope.maptype) {
   case ...
}

Reply:
I just have concern about "doNormalization" and "doSentiment". Only countmap needs them. E.g. when we first switch from countmap to pointmap, drawCountMap will be called because new doNormalization result != old result. Also, heatmap and pointmap may require their own additional checks in the future.